server/src/main/java/org/cloudfoundry/identity/uaa/account/PasswordChangeEndpoint.java

```diff
@@ -59,12 +60,15 @@
     private HttpMessageConverter<?>[] messageConverters = new RestTemplate().getMessageConverters().toArray(
                     new HttpMessageConverter<?>[0]);
 
+    private ResourcePropertySource messages;
+
     /**
      * Constructor.
      *
      * @param passwordHistoryRestriction new passwords must be different than this number of previous passwords to be considered valid
      */
-    public PasswordChangeEndpoint(final int passwordHistoryRestriction) {
+    public PasswordChangeEndpoint(final ResourcePropertySource messages, final int passwordHistoryRestriction) {
+        this.messages = messages;
```

Comment:
I removed the copy-paste of the error message to a message file (following existing examples).